File: build_site.py
"""Build the flat Qamelot page collection for Sites hosting."""

from __future__ import annotations

import shutil
from pathlib import Path

ROOT = Path(__file__).resolve().parent
DIST = ROOT / "dist"
CLIENT = DIST / "client"
SERVER = DIST / "server"

if DIST.exists():
    shutil.rmtree(DIST)
CLIENT.mkdir(parents=True)
SERVER.mkdir(parents=True)

for page in ROOT.glob("*.html"):
    shutil.copy2(page, CLIENT / page.name)
for extension in ("*.mp3", "*.png", "*.ico", "*.svg"):
    for asset in ROOT.glob(extension):
        shutil.copy2(asset, CLIENT / asset.name)
shutil.copy2(ROOT / "page-manifest.json", CLIENT / "page-manifest.json")

(SERVER / "index.js").write_text(
    '''export default {
  async fetch(request, env) {
    const url = new URL(request.url);
    if (url.pathname === "/") url.pathname = "/index.html";
    if (!env.ASSETS?.fetch) {
      return new Response("Static asset service unavailable", { status: 500 });
    }
    return env.ASSETS.fetch(new Request(url, request));
  }
};
''',
    encoding="utf-8",
    newline="\n",
)
print(f"Built {len(list(CLIENT.glob('*.html')))} webpages")

File: build-site.mjs
import { cp, mkdir, readdir, rm } from "node:fs/promises";

await rm("dist", { recursive: true, force: true });
await mkdir("dist/client", { recursive: true });
await mkdir("dist/server", { recursive: true });
const files = await readdir(".");
for (const file of files.filter((name) => name.endsWith(".html"))) {
  await cp(file, `dist/client/${file}`);
}
await cp("page-manifest.json", "dist/client/page-manifest.json");
await cp("n8-tools-transformers.css", "dist/client/n8-tools-transformers.css");
await cp("n8-tools-transformers.js", "dist/client/n8-tools-transformers.js");
await cp("funpromptz-og.png", "dist/client/funpromptz-og.png");
await cp("n8-icon.png", "dist/client/n8-icon.png");
await cp("loc-me-in", "dist/client/loc-me-in", { recursive: true });
await cp("worker/index.js", "dist/server/index.js");
